### PR TITLE
Lib Loader Radiobutton

### PR DESCRIPTION
**Pull request by Megamouse**
This moves the "auto load" and "liblv2 only" options to a radiobutton and adds "load manually" and "auto+manually"

preview

### CODE REVIEW

**Comment by Megamouse** [comment]
still a bug: list is always enabled at first on start even if auto or liblv2 is selected

**Comment by Megamouse** [comment]
should be fixed now. sorry for the bugfixes

**Comment by SakataGintokiYT** [comment]
@al0xf 
This pull request is unnecessary.

**Comment by al0xf** [comment]
No. This is an important usability improvement. It is literally very necessary.

**Comment by Zangetsu38** [comment]
yes, i think also, is more clear for user like this

File: rpcs3/Emu/Cell/PPUModule.cpp [hunk 1]
```diff
@@ -1307,7 +1320,7 @@ void ppu_load_exec(const ppu_exec_object& elf)
 	ppu->cmd_push({ppu_cmd::initialize, 0});
 
 	// TODO: adjust for liblv2 loading option
-	if (!g_cfg_load_liblv2)
+	if (g_cfg_lib_loader.get() != lib_loader_mode::manual)
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/PPUModule.cpp]: shouldn't this say `if (g_cfg_lib_loader.get() != lib_loader_mode::liblv2only)` ? o.O what a mistake

**Comment by al0xf** [comment]
@Megamouse While you are at it: rename "D3D adapter" to "DirectX12 adapter" or "D3D adapter (DirectX12 only)" or similar. Yes I know that DX12 != D3D and I don't care. This is easier for people. 

Moreover, also rename the menu option "PAD Settings" to "Keyboard settings".

**Comment by Megamouse** [comment]
Why not grey out the button if other than directX is selected?
Keyboard Settings or Input Settings?
I don't really know what liblv2 does except for what it says :)

**Comment by al0xf** [comment]
@Megamouse Unfortunately there is a RPCS3 set up video on YouTube with 100k views where people are instructed to set their GPU in that list (yes the video is dead wrong) and a lot of people ask about it due to that.

**Comment by AniLeo** [comment]
@Megamouse don't, packages are for more than games, that will generate more confusion

**Comment by AniLeo** [comment]
Maybe have the radio buttons be inside Load libraries, below the searchbox.
Load automatically + manually to "Automatic + manual selection load"

**Comment by paulsapps** [comment]
"Load automatically + manually" makes no sense to me :D

**Comment by twdarkeh** [comment]
"Load Automatically + manually selected" sounds better?

**Comment by Megamouse** [comment]
maybe "Load automatic + manual selection" ? or "Auto-load libraries + manual selection"
if english hadn't that ally or ly at the end of an adverb it would be easier :)

**Comment by komawoyo** [comment]
I prefer: 
- Auto Load Required Libraries
- Auto Load + Manual Select Libraries
- Manual Select Libraries
- Load libvl2.sprx Only

**Comment by twdarkeh** [comment]
"Controller Settings" is wrong though, since it only works for keyboards. "Keyboard Settings" is best for that.

**Comment by komawoyo** [comment]
If that's case then this section in settings will be confusing

image

**Comment by twdarkeh** [comment]
Confusing or not, the PAD Settings option ONLY works for remapping keyboard controls; you cannot remap a controller at this time. Thus the option should reflect that.

**Comment by kd-11** [comment]
What does load automatically + manually actually mean/do? What is its current equivalent?

**Comment by Talkashie** [comment]
It does the same thing as "load required libraries" and "only load libv2.sprx", but it is much more straightforward.

**Comment by kd-11** [comment]
Is it though? A new user will have no understanding of liblv2 so telling them that an option does both manual and auto selection makes no sense. Also liblv2 is not exactly manual selection. I mean, the option says that only liblv2 is to be loaded exclusively.
For liblv2 option, a checklist makes more sense imo. I'm a contributor and even I had to have an option explained to  me so I dont think its easier for noobs this way.

**Comment by Talkashie** [comment]
I believe the way it is set up right now makes it so that if you have "load required libraries" enabled, "liblv2 only" will add additional libraries on top of automatic ones. I am not sure if I am missing your question or not... sorry if this isn't what you are talking about.

**Comment by kd-11** [comment]
Its my question, but in that case, isnt liblv2 automatically choosing the modules by itself? The cellXXX modules in rpcs3 are actually just liblv2 HLE btw. In that case liblv2 is still not manual selection (selection from the list box)

**Comment by Talkashie** [comment]
I swear I was told differently a while back. Plus, I'm pretty sure if you have both the automatic option and libv2.sprx enabled, choosing libraries from the list box will have different effects, as if they are being loaded.

**Comment by Megamouse** [comment]
"liblv2 only" and "autoload only" (if you wish) exclude the manual selection.
but in the (current) state, you can choose both and get only manual selection afaik. therefore the 4 options

**Comment by kd-11** [comment]
Afaik liblv2 will override whatever auto/manual module selection you have, so I think you only need 3 options:
- Auto load
- Manual load
- liblv2 load (its a different type of auto load)
You should consult on this; I might be wrong/misinformed since I havent gone over the code for recent improvements.

**Comment by Megamouse** [comment]
you can add manual on auto.
In order to prevent user from choosing manually from the get go, auto is set to default with manual disabled

**Comment by kd-11** [comment]
Adding manual on auto should do nothing afaik. As long as auto load is enabled, whatever you select manually is ignored.

**Comment by Megamouse** [comment]
unbenannt

Neko even said later I could do 4 options

**Comment by raven02** [comment]
@Megamouse , are they any games actually need the "Adding manual on auto" ? If yes , then we need it and vice versa.

**Comment by kd-11** [comment]
We need a better name for the auto+manual option. Some users have a hard time understanding seemingly obvious things like the d3d adapter and directx12 (ironically part of this pr). I fear some will select all the manual libs and complain that a game doesnt work because they selected automatic loading.

**Comment by Megamouse** [comment]
then lets add (recommended) to one of the options.
you can't blame the devs if user chooses to ignore common sense.

File: rpcs3/Emu/Cell/PPUModule.cpp [hunk 2]
```diff
@@ -119,9 +119,23 @@ LOG_CHANNEL(sysPrxForUser);
...
+{
+	automatic,
+	manual,
+	both,
+	liblv2only
+};
+
+cfg::map_entry<lib_loader_mode> g_cfg_lib_loader(cfg::root.core, "Lib Loader", 1,
+{
+	{ "Automatically load required libraries ", lib_loader_mode::automatic },
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/PPUModule.cpp]: Space

**Comment by AlbertAdair** [comment]
# **_Hello, you could create an option to select the user or create a user because in the Game Minecraft RPCS3 asks for access to the UserList for another player to join but as there is no option please you could create that option?_**

**Comment by AniLeo** [comment]
Don't use headers incorrectly, everyone else uses normal sized letters. 

Support for multiple ps3 users is on the roadmap but as a long term goal, unless someone wants to do it.

**Comment by AlbertAdair** [comment]
AniLeo Sorry I'm new and I will not write again, okay

**Comment by AlbertAdair** [comment]
I do not know anything about programming but if I learn I could make the user option how could someone teach me?

**Comment by hcorion** [comment]
@AlbertAdair 
RPCS3 is a highly complicated program that some mature programmers (like 4-5 years) would struggle with. 

If you're really serious about learning to program, find some online tutorials, I'd recommend the python programming language: https://www.learnpython.org/ 

but please don't reply to this message, we've all got our own lives to attend to, and most of us don't have the time to devote to helping you learn programming.

**Pull request merged**